src/gc/gcinterface.h

```diff
@@ -444,6 +444,12 @@ class IGCHandleManager {
     virtual void DestroyHandleOfUnknownType(OBJECTHANDLE handle) = 0;
 
     virtual void* GetExtraInfoFromHandle(OBJECTHANDLE handle) = 0;
+
+    virtual void StoreObjectInHandle(OBJECTHANDLE handle, Object* object) = 0;
+
+    virtual bool StoreObjectInHandleIfNull(OBJECTHANDLE handle, Object* object) = 0;
+
+    virtual Object* CompareAndSwapObjectInHandle(OBJECTHANDLE handle, Object* object, Object* comparandObject) = 0;
```

Comment:
Interlocked CompareExchange is also name of the cross-plat .NET API, and we have `Interlocked ::CompareExchange` in gcenv.interlocked.inl.

The GCHeap interface has a Windows feel in it because of it is using CamelCase naming, so I do not think that name with Windows heritage looks unnatural in it.

BTW: C++ standard is calling it `atomic::compare_exchange` (http://en.cppreference.com/w/cpp/atomic/atomic).